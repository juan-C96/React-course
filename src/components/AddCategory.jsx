import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( { onNewCategory } ) => {

    const [inputValue, setinputValue] = useState("");

    const onInputChange = ({ target }) => {
        setinputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if ( inputValue.trim().length >= 1 ) {
            //setCategory( categorias => [inputValue, ...categorias] );
            onNewCategory( inputValue.trim() );
            setinputValue('');
        }
    }

    return (
        <form onSubmit = { onSubmit } aria-label="form">
            <input
                type="text"
                placeholder="Buscar gif"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}


AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}