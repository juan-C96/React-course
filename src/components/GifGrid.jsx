import { GifGridItem } from './GifGridItem';
import {useFetchGifs} from '../hooks/useFetchGifs'
import PropTypes from 'prop-types';

export const GifGrid = ({ categoria }) => {

    const { images, isLoading } = useFetchGifs( categoria );

    return (
        <>
            <h3>{categoria}</h3>
            {
                isLoading && (<h2>Cargando...</h2>)
            }

            <div className='card-grid'>
                {
                    images.map( image => (
                        <GifGridItem 
                        key={ image.id } 
                        { ...image }/>
                        /*title={ image.title }
                        url={ image.url }/>*/
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    categoria: PropTypes.string.isRequired,
}
