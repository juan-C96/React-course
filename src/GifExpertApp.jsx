import { useState } from "react";
import { AddCategory, GifGrid } from './components'

export const GifExpertApp = () => {

    const [categorias, setCategorias] = useState([]);

    const onAddCategory = (newCategory) => {

        if (categorias.includes(newCategory)) return;
        //categorias.push('event');
        console.log(newCategory)
        //setCategorias( cat => [ ...cat, 'Valorant'] );
        setCategorias([newCategory, ...categorias]);

    };

    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory
                //setCategory = { setCategorias } 
                onNewCategory={onAddCategory}
            />

            {categorias.map(category => (
                <GifGrid key={category} categoria={category} />
            )
            )}
        </>
    )
}
