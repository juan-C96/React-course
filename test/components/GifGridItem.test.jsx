import { render, screen } from "@testing-library/react"
import { GifGridItem } from '../../src/components';


describe('Pruebas al componente GifGridItem', () => {

    const title = '123';
    const url = 'http://localhost/VVVVV';

    test('Debe hacer match con el Snapshot', () => {
        const { container } = render(<GifGridItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar la imagen con el url y el alt indicado', () => {
        render(<GifGridItem title={title} url={url} />);
        /*/expect( screen.getByRole( 'img' ).src ).toBe( url );
        expect( screen.getByRole( 'img' ).alt ).toBe( title );*/
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Debe mostrar el titulo en el componente', () => {
        render(<GifGridItem title={ title } url={ url }/>);
        expect( screen.getByText( title ) ).toBeTruthy();

    })
})