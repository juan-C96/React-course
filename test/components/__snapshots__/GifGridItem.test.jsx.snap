// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente GifGridItem Debe hacer match con el Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="123"
      src="http://localhost/VVVVV"
    />
    <p>
      123
    </p>
  </div>
</div>
`;
